keys/keys.go

```diff
@@ -27,6 +28,49 @@ import (
 // PrivKeyBytesLen are 32-bytes for all supported curvetypes
 const PrivKeyBytesLen = 32
 
+// ImportPrivKey returns a Keypair from a hex-encoded privkey string
+func ImportPrivKey(privKeyHex string, curve types.CurveType) (*KeyPair, error) {
```

Comment:
I'd recommend using `keys.PrivateKey` here because it already has curve and does the hex decoding for you!

Reply:
JK not possible, we don't have a `keys.PrivateKey` type.